plugins/ptp_operator/metrics/manager.go

```diff
@@ -261,10 +261,19 @@ func (p *PTPEventManager) PublishEvent(state ptp.SyncState, ptpOffset int64, sou
 		log.Infof("PublishEvent state=%s, ptpOffset=%d, source=%s, eventType=%s", state, ptpOffset, source, eventType)
 		return
 	}
+
 	// /cluster/xyz/ptp/CLOCK_REALTIME this is not address the event is published to
 	data := p.GetPTPEventsData(state, ptpOffset, source, eventType)
 	resourceAddress := fmt.Sprintf(p.resourcePrefix, p.nodeName, string(p.publisherTypes[eventType].Resource))
 	p.publish(*data, resourceAddress, eventType)
+	// publish the event again as overall sync state
+	// SyncStateChange is the overall sync state including PtpStateChange and OsClockSyncStateChange
+	if eventType == ptp.PtpStateChange || eventType == ptp.OsClockSyncStateChange {
```

Comment:
you will end up sending twice or more time . when there is ptpState Change event in OC  or BC, there will be following OS clock event , not the other way .
So 1) send overall state when there is only OSClock for OC/BC and 2) send overall State for BC without Phc2sys in its Profile.
3) Also for HA ,send Over All state when OSCLOCK sends event
FO T_-GM it is handled differently  OSClock sate don't impact T-GM state but tsphc and DPLL impacts T-GM state

Reply:
updated